go/src/koding/klient/machine/index/index.go

```diff
@@ -195,25 +232,47 @@ func (idx *Index) Lookup(name string) (*Node, bool) {
 // Compare rereads the given file tree roted at root and compares its entries
 // to previous state of the index. All detected changes will be stored in
 // returned Change slice.
-func (idx *Index) Compare(root string) (cs ChangeSlice) {
+func (idx *Index) Compare(root string) ChangeSlice {
+	return idx.CompareBranch("", root)
+}
+
+// CompareBranch rereads the given file tree roted at root and compares its entries
+// with index state roted at branch node.
+//
+// All detected changes will be stored in returned Change slice.
+// If branch is empty, the comparison is made against root of the index.
+func (idx *Index) CompareBranch(branch, root string) (cs ChangeSlice) {
```

Comment:
Oh, this way, yeah. I'll refactor this later as there's too much code outside this PR that is dependant on this.